resources/shellHelperFunctions.sh

```diff
@@ -304,4 +304,30 @@ _select_node_version_with_nvm() {
 
 check-markdown() {
   node "$KEYMAN_ROOT/resources/tools/check-markdown" --root "$1"
+}
+
+builder_do_typescript_tests() {
```

Comment:
Good refactoring.

Some questions about maintenance:

1. Could we add a short preamble to note what this does?
2. From a maintenance view, the `builder_` prefix would make me assume this function is defined in `builder.inc.sh` with all the other `builder_` functions.

Reply:
> 1. Could we add a short preamble to note what this does?

Yes, will do.

> 2. From a maintenance view, the `builder_` prefix would make me assume this function is defined in `builder.inc.sh` with all the other `builder_` functions.

I didn't want to put this into the main builder script, as it is specific to Typescript (and perhaps specific even to Developer use of Typescript; not sure it would transplant to /web for example). Ideally, we may have a set of builder sub-scripts, e.g. builder.typescript.inc.sh, builder.meson.inc.sh, etc, which we can draw from, and I would like to consider that for 19.0 (perhaps low pri though). shellHelperFunctions.sh is a bit of a random grab bag of functions that need refactoring and renaming, and eventually it should go away.

Reply:
builder.typescript.inc.sh etc feat documented in #12744